### ISSUE TITLE
toast assertive

### ISSUE THREAD

**Issue by rober423** [materializecss/materialize#301]
My [original request on the Dogfalo repo](https://github.com/Dogfalo/materialize/issues/6672) was about Toast messages being read by screenreader when they appear.

Readability is the first and most important thing for accessibility. See https://sheribyrnehaber.medium.com/designing-toast-messages-for-accessibility-fb610ac364be#bd14

UPDATE:
I see that you have added role="alert" and it's related attributes here. Thank you!
https://github.com/materializecss/materialize/blob/2f6a0b0c7d66ef5ffa46b2214ff136559a73c339/js/toasts.js#L198

I do just want to ask though, why choose aria-live='assertive'? Is it expected of Materialize users to use toast for only very important messages? Or would it be possible for users to have the option to change it to 'polite', with the default being 'assertive'?

**Comment by wuda-io** [comment]
Thanks for your input! I read the specs of this aria attributes quickly and did not really put much effort in thinking for possible alternatives.

Yesterday it was included into the project. I dont think if we make an additional option that many users will understand or config it correctly.

What are other possiblities for assertive and what is the advantage with this method?
I think alert was ok. Please elaborate in more detail what you suggest or even better make a PR then we can go and pick the optimal solutions which fits most needs.

**Comment by rober423** [comment]
The only options for aria-live are assertive, off, or polite, as described [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-live). Materialize users may want to use assertive when their toast is important, such as an error message. Alternatively, they may want to use polite to delay announcement until whatever is currently being read is finished, such as for a status message. I might make a pull request and attempt to make the aria-live option.

**Comment by wuda-io** [comment]
Yes that sounds awesome! I have to admit, that you know better about the details of this feature, every help is greatly appreciated. Thanks 🙏

**Comment by wuda-io** [comment]
Another issue would be here, that the toast is only visible for a short amount of time,
so how is this handled then? If the toast dissapears in the time reading some other text,
maybe the state changed... Alert makes sense, but thats a very specific discussion, so I dont know what is better.
Maybe @mauromascarenhas  can elaborate why alert was used here.

**Comment by rober423** [comment]
No it is handled fine. If any aria-live element gets a child added, that child's text gets added to a queue in the screenreader. so even if the element disappears, the screenreader will still read the alert.

**Comment by rober423** [comment]
currently working on a pull request. For a better explanation of aria-live please see [https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions#live_regions](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions#live_regions)

**Comment by mauromascarenhas** [comment]
> Another issue would be here, that the toast is only visible for a short amount of time, so how is this handled then? If the toast dissapears in the time reading some other text, maybe the state changed... Alert makes sense, but thats a very specific discussion, so I dont know what is better. Maybe @mauromascarenhas can elaborate why alert was used here.

I've actually considered 2 different scenarios:

1. The notification is important, but no longer valid for the current context (e.g.: The user changes routes while navigating through a SPA, possibly making the notification either redundant or completely unnecessary (think of "You are about to be redirected"));
2. Usually, a toast is emmited either when a important action is performed or when an event is triggered, then the user must receive an immediate feedback.

However, considering a low priority scenario, as described in previous comments, a new "politeness" option could be used to set the alert in the reader's queue with a default value, since people generally do not even care about setting ARIA attributes.

**Comment by rober423** [comment]
> No it is handled fine. If any aria-live element gets a child added, that child's text gets added to a queue in the screenreader. so even if the element disappears, the screenreader will still read the alert.

@mauromascarenhas & @wuda-io, I apologize. This statement is incorrect. In testing for the changes I made, I realized that the screenreader does not read an alert if it has been removed from DOM.

If you test my branch with a screenreader such as NVDA, you'll see/hear that pressing the polite toast button focuses the other new button I made (which has sufficiently long text to help you notice the alert is read _after_ the screenreader finishes that text), however the alert is not announced still since it disappears. I tried making the DOM removal get delayed, but that didn't work either (https://github.com/materializecss/materialize/pull/302/commits/9d43b2e41e92bc877ed13878f24235aadee29a07).

Considering this now seems to be more effort than it's worth, I'm not going to investigate any further. It is probably fine the way it is, which assumes that toasts should be read immediately.
